src/app/core/access-checker/cache/cache-refresh.service.js

```diff
@@ -1,44 +1,29 @@
 'use strict';
 
-const
-	q = require('q'),
-
-	deps = require('../../../../dependencies'),
+const deps = require('../../../../dependencies'),
 	dbs = deps.dbs,
 	logger = deps.logger,
 
 	accessChecker = require('../access-checker.service'),
 	CacheEntry = dbs.admin.model('CacheEntry');
 
-
-module.exports.run = function(svcConfig) {
+module.exports.run = async (svcConfig) => {
 	logger.debug('Access Checker: Checking cached users...');
 
-	const refresh = svcConfig.refresh || 8*3600000; // default to 12 hours
-
-	// Create a defer for the response
-	const defer = q.defer();
+	const refresh = svcConfig.refresh || 8*3600000; // default to 8 hours
 
 	// Find all the keys that need to be refreshed
-	CacheEntry.find({ ts: { $lt: Date.now() - refresh } }).exec((error, results) => {
-		if(null != error) {
-			defer.reject(error);
-		} else {
-			if(results.length > 0) {
-				logger.info('Access Checker: Refreshing %s users', results.length);
-			}
-
-			// Iterate through each object, refreshing as you go
-			const defers = [];
-			results.forEach((e) => {
-				logger.debug('Access Checker: Refreshing %s', e.key);
-				defers.push(accessChecker.refreshEntry(e.key));
-			});
-
-			q.all(defers).then(defer.resolve, defer.reject).done();
-		}
+	const results = await CacheEntry.find({ ts: { $lt: Date.now() - refresh } }).exec();
```

Comment:
Do we need to wrap this in a try/catch in the case of a failure?

Reply:
I _think_ the async/await would handle this by rejecting the promise with the error, based on https://javascript.info/async-await#:~:text=If%20a%20promise%20resolves%20normally,throw%20statement%20at%20that%20line. If we'd prefer the try/catch approach for clarity, I'm certainly open to it!

Reply:
Seems reasonable! Based on the awaits I saw in here, most if not all were wrapped, so I didn't know if we needed it or not. 
Looks like this should follow the same signature of the Promise.all, rejecting the first promise that is rejected, so we shouldn't need special handling of this function. Probably fine to keep it as-is.